tests/debug-tests/send.py

```diff
@@ -32,31 +31,28 @@ def server(env, port, func):
     async def f(listener_port):
         # coroutine shows up when the client asks
         # to connect
+        set_rmm()
+
         async def write(ep):
 
             print("CREATING CUDA OBJECT IN SERVER...")
             cuda_obj_generator = cloudpickle.loads(func)
             cuda_obj = cuda_obj_generator()
             msg = {"data": to_serialize(cuda_obj)}
             frames = await to_frames(msg, serializers=("cuda", "dask", "pickle"))
-            for i in range(ITERATIONS):
-                # Send meta data
-                await send(ep, frames)
-
-                frames, msg = await recv(ep)
+            while True:
+                for i in range(ITERATIONS):
+                    print("ITER: ", i)
+                    # Send meta data
+                    await send(ep, frames)
 
-            print("CONFIRM RECEIPT")
-            close_msg = b"shutdown listener"
-            msg_size = np.empty(1, dtype=np.uint64)
-            await ep.recv(msg_size)
+                    frames, msg = await recv(ep)
 
-            msg = np.empty(msg_size[0], dtype=np.uint8)
-            await ep.recv(msg)
-            recv_msg = msg.tobytes()
-            assert recv_msg == close_msg
-            print("Shutting Down Server...")
-            await ep.close()
-            lf.close()
+                print("CONFIRM RECEIPT")
+                await ep.close()
```

Comment:
I only see the bug when when we try to send using a closed endpoint. Exiting the loop after closing the endpoint avoids the bug:
```suggestion
                await ep.close()
                break
```
However, exiting the loop via. a `ucp.exceptions.UCXCloseError` should be perfectly legal.

Reply:
Hmm, in this test, I wanted to keep the sender alive forever.   Breaking here will close the sender, right ?

Reply:
it breaks in the next `send()` call anyways because `ep` is closed.